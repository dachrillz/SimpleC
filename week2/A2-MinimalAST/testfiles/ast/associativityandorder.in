int a(){
    a = 5 + 5 + 5;
    return;
}

int b(){
    a = 5*5+1;
    return;
}

int c(){
    a = 1+5*5;
    return;
}

int d(){
    a = (1+2)+3;
    b = 1+2+3;
    c = 2+(2+3);
    return;
}

int e(){
    a=1+2*3;
    b=(1+2)*3;
    return;
}

int f(){
    a=1+2-3*10/5;
    return;
}
