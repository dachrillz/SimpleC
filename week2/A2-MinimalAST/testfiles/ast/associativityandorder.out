Program
  List
    Function ID="a"
      null
      List
        IdUse ID="a"
          Plus
            Plus
              Var ID="5"
              Var ID="5"
            Var ID="5"
        Return
          null
    Function ID="b"
      null
      List
        IdUse ID="a"
          Plus
            Mul
              Var ID="5"
              Var ID="5"
            Var ID="1"
        Return
          null
    Function ID="c"
      null
      List
        IdUse ID="a"
          Plus
            Var ID="1"
            Mul
              Var ID="5"
              Var ID="5"
        Return
          null
    Function ID="d"
      null
      List
        IdUse ID="a"
          Plus
            Plus
              Var ID="1"
              Var ID="2"
            Var ID="3"
        IdUse ID="b"
          Plus
            Plus
              Var ID="1"
              Var ID="2"
            Var ID="3"
        IdUse ID="c"
          Plus
            Var ID="2"
            Plus
              Var ID="2"
              Var ID="3"
        Return
          null
    Function ID="e"
      null
      List
        IdUse ID="a"
          Plus
            Var ID="1"
            Mul
              Var ID="2"
              Var ID="3"
        IdUse ID="b"
          Mul
            Plus
              Var ID="1"
              Var ID="2"
            Var ID="3"
        Return
          null
    Function ID="f"
      null
      List
        IdUse ID="a"
          Minus
            Plus
              Var ID="1"
              Var ID="2"
            Div
              Mul
                Var ID="3"
                Var ID="10"
              Var ID="5"
        Return
          null
