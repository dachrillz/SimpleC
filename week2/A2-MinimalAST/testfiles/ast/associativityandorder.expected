Program
  List
    Function ID="a"
      List
        IdUse ID="a"
          Plus
            Plus
              Numeral NUMERAL="5"
              Numeral NUMERAL="5"
            Numeral NUMERAL="5"
        Return
    Function ID="b"
      List
        IdUse ID="a"
          Plus
            Mul
              Numeral NUMERAL="5"
              Numeral NUMERAL="5"
            Numeral NUMERAL="1"
        Return
    Function ID="c"
      List
        IdUse ID="a"
          Plus
            Numeral NUMERAL="1"
            Mul
              Numeral NUMERAL="5"
              Numeral NUMERAL="5"
        Return
    Function ID="d"
      List
        IdUse ID="a"
          Plus
            Plus
              Numeral NUMERAL="1"
              Numeral NUMERAL="2"
            Numeral NUMERAL="3"
        IdUse ID="b"
          Plus
            Plus
              Numeral NUMERAL="1"
              Numeral NUMERAL="2"
            Numeral NUMERAL="3"
        IdUse ID="c"
          Plus
            Numeral NUMERAL="2"
            Plus
              Numeral NUMERAL="2"
              Numeral NUMERAL="3"
        Return
    Function ID="e"
      List
        IdUse ID="a"
          Plus
            Numeral NUMERAL="1"
            Mul
              Numeral NUMERAL="2"
              Numeral NUMERAL="3"
        IdUse ID="b"
          Mul
            Plus
              Numeral NUMERAL="1"
              Numeral NUMERAL="2"
            Numeral NUMERAL="3"
        Return
    Function ID="f"
      List
        IdUse ID="a"
          Minus
            Plus
              Numeral NUMERAL="1"
              Numeral NUMERAL="2"
            Div
              Mul
                Numeral NUMERAL="3"
                Numeral NUMERAL="10"
              Numeral NUMERAL="5"
        Return
